### ISSUE TITLE
pull features from angular2 seed

### ISSUE THREAD

**Issue by lathonez** [lathonez/clicker#19]
See what https://github.com/mgechev/angular2-seed have done since we forked and steal any cool stuff :)

**Comment by lathonez** [comment]
Had a quick look tonight, they have the following:

* painless adding of node_modules (e.g. no karma.conf fuckery)
* test watching (dunno if this is worth it with the boilerplate overhead, will see)
* e2e testing with protractor

Ahead of this I've refactored gulpfile.js (now ts!) along the same lines as theirs.

**Comment by lathonez** [comment]
Taken gulp-inline-ng2-template in 0b6f152e045aafb460d6401821e9133e8a37f76f, so we don't need to copy html and need less proxying

**Comment by lathonez** [comment]
When I include moment.js I still get a 404 from Karma using angular2 seed.

**Comment by lathonez** [comment]
As luck would have it, they've had the exact same issue raised there:

https://github.com/mgechev/angular2-seed/issues/468

So they're using system config instead of proxying in Karma. Seems a bit cleaner..

**Comment by lathonez** [comment]
They do in config.ts:

```typescript
// ----------------
// SystemsJS Configuration.
const SYSTEM_CONFIG_DEV = {
  defaultJSExtensions: true,
  paths: {
    [BOOTSTRAP_MODULE]: `${APP_BASE}${BOOTSTRAP_MODULE}`,
    'angular2/*': `${APP_BASE}angular2/*`,
    'rxjs/*': `${APP_BASE}rxjs/*`,
    '*': `${APP_BASE}node_modules/*`
  },
  packages: {
    angular2: { defaultExtension: false },
    rxjs: { defaultExtension: false }
  }
};

export const SYSTEM_CONFIG = SYSTEM_CONFIG_DEV;
```

And then this in index.html:

```html
  <% if (ENV === 'dev') { %>
  <script>System.config(<%= JSON.stringify(SYSTEM_CONFIG) %>)</script>
  <% } %>
```

which is built as part of build.dev.index:

```
export = function buildIndexDev(gulp, plugins) {
  return function () {
    return gulp.src(join(APP_SRC, 'index.html'))
      .pipe(inject('shims'))
      .pipe(inject('libs'))
      .pipe(inject())
      .pipe(plugins.template(templateLocals()))
      .pipe(gulp.dest(APP_DEST));
  };
```

**Comment by lathonez** [comment]
Lot of flapping, but it might help us get away from stuff like [this](https://github.com/lathonez/clicker/blob/master/test/ionic-angular.js)

**Comment by lathonez** [comment]
Seems like a bit of a rabbit hole considering it doesn't give you external deps for free, dropping

**Comment by lathonez** [comment]
test debug watching working & written up

**Comment by lathonez** [comment]
Closed in favour of #38
